### DISCUSSION TITLE
Button Groups for one too many actions required in one row.

### DISCUSSION THREAD

**Discussion by LogicSatinn** [Power-Components/livewire-powergrid#1057] [Ideas]
# ⚡ PowerGrid ⚡ Feature Request

## Summary
More than once, my team and I found ourselves adding so many buttons for different actions on one row, and this makes everything look weird and cluttered. We used some unconventional, weird ways to fix this which left us desiring something cleaner. This issue asks for a button group feature, that resembles that of FilamentPHP where we can group row actions without making a mess of it.

### Brief description of what is at stake.
It might need a lot of work. Adding support for both Tailwind and Bootstrap 5. Any contribution towards this would just be some addition on top. I don't think there will be any breaking changes due to the addition of this feature.

**Comment by luanfreitasdev** [comment]
Hello, can you give more examples and details of how we could implement it in Powergrid?. Thanks

**Reply by LogicSatinn** [reply]
Well ... SInce we are grouping buttons together, that means we can create a class that takes a list (array) of Buttons, just as they are right now. Then there could be a custom blade file for these grouped buttons, ideally a dropdown in my opinion. 

Here's an example of what I'm thinking.
![image](https://github.com/Power-Components/livewire-powergrid/assets/69092766/0c5a7c4b-66c0-4f34-b081-065736bb5aa5)

Code snippet could look like this

```
ButtonGroup::make([
  Button::add('View on Storefront')
               ->caption('View on Storefront'),

 ...

 Button::add('delete')
             ->caption('Delete Product')
            ->class('text-red-600')
            ->method('delete')
]);
```

**Reply by elsoftpy** [reply]
I think this is a good idea. 

I have a use case for this. Let's say we have a Project with tasks, these tasks can be assigned to users, then each user should be able to mark the task as done (clicking an action button in the table).

The project manager should also be able to mark other users' tasks as done, but in order to do so the manager should have one button per task. Grouping these buttons as suggested would be a great solution.
